facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -94,6 +105,21 @@ public ServiceFacade(D legacySvc, D lightblueSvc, String implementationName, Pro
 
         timeoutConfiguration = new TimeoutConfiguration(DEFAULT_TIMEOUT_MS, implementationName, this.properties);
 
+        // create ThreadPoolExecutor for all facades
+        synchronized("executor") {
+            if (executor == null) {
+                int threadPoolSize = Integer.parseInt(this.properties.getProperty(CONFIG_PREFIX+implementationName+".threadPool.size", "50"));
+
+                // cached means automatic thread reclamation
+                ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) Executors.newCachedThreadPool();
+                // fixed size, no queuing
+                threadPoolExecutor.setCorePoolSize(threadPoolSize);
+                threadPoolExecutor.setMaximumPoolSize(threadPoolSize);
+                executor = MoreExecutors.listeningDecorator(threadPoolExecutor);
+                log.info("Initialized facade threadPool.size="+threadPoolSize);
```

Comment:
You can have multiple facaded services in a single deployment, but only one pool, so it's not there is no 1 to 1 assoication between the pool and implementationName. But perhaps it's still better than nothing.